apps/api/src/app/workflows-v2/util/template-parser/liquid-parser.ts

```diff
@@ -0,0 +1,126 @@
+import { Template, Liquid, RenderError, LiquidError } from 'liquidjs';
+import { TemplateParseResult, InvalidVariable } from './parser-types';
+import { isValidTemplate, extractLiquidExpressions } from './parser-utils';
+
+const LIQUID_CONFIG = {
+  strictVariables: true,
+  strictFilters: true,
+  greedy: false,
+  catchAllErrors: true,
+} as const;
+
+/**
+ * Copy of LiquidErrors type from liquidjs since it's not exported.
+ * Used to handle multiple render errors that can occur during template parsing.
+ * @see https://github.com/harttle/liquidjs/blob/d61855bf725a6deba203201357f7455f6f9b4a32/src/util/error.ts#L65
+ */
+class LiquidErrors extends LiquidError {
+  errors: RenderError[];
+}
+
+function isLiquidErrors(error: unknown): error is LiquidErrors {
+  return error instanceof LiquidError && 'errors' in error && Array.isArray((error as LiquidErrors).errors);
+}
+
+/**
+ * Parses a Liquid template string and extracts all variable names, including nested paths.
+ * Validates the syntax and separates valid variables from invalid ones.
+ *
+ * @example
+ * // Valid variables
+ * parseLiquidVariables('Hello {{user.name}}, your score is {{user.score}}')
+ * // Returns:
+ * {
+ *   validVariables: ['user.name', 'user.score'],
+ *   invalidVariables: []
+ * }
+ *
+ * @example
+ * // Mixed valid and invalid syntax
+ * parseLiquidVariables('{{user.name}} {{invalid..syntax}}')
+ * // Returns:
+ * {
+ *   validVariables: ['user.name'],
+ *   invalidVariables: [{
+ *     context: '>> 1| {{invalid..syntax}}\n                ^',
+ *     message: 'expected "|" before filter',
+ *     variable: '{{invalid..syntax}}'
+ *   }]
+ * }
+ *
+ * @param template - The Liquid template string to parse
+ * @returns Object containing arrays of valid and invalid variables found in the template
+ */
+export function extractLiquidTemplateVariables(template: string): TemplateParseResult {
+  if (!isValidTemplate(template)) {
+    return { validVariables: [], invalidVariables: [] };
+  }
+
+  const liquidRawOutput = extractLiquidExpressions(template);
```

Comment:
@djabarovgeorge, Can you please explain why we use extra expressions using regex before passing them to Liquid? I'd expect Liquid to get the raw string and do the calculations.

Reply:
Awesome question, this is the way we can provide `valid` and `invalid`.
because once we parse we get an exception.
we made a workaround it, so now we will parse every liquid js Output separately. 
That will allow as to return both `valid` and `invalid` variables.
this one is a bit complex, we can sync about it if needed.

Reply:
Thanks for the explanation.